### ISSUE TITLE
Physical barrier

### ISSUE THREAD

**Issue by jetgeo** [ERTICO-TN-ITS/TN-ITS-Open#21]
Suggested by Sweden, ref https://github.com/ERTICO-TN-ITS/TN-ITS-Open/issues/9

The is a feature to indicate "Closed to all vechicle in all directions".
TN-ITS requires a location referencing but type is flexible, ie can be both linear or
point feature.
This could be added as "Closed to all vechicle in all directions", point feature.
Can be complicated to make all read behind it as "restricted access" as the network
can be extensive,
A physical barrier, typically locked at the entrance of a private road to the forest.
Here with the sign for "mo notorised vehicles execept for those with permission"

![image](https://user-images.githubusercontent.com/7501876/153869263-ba1aa762-f6f3-45bb-a644-80687ea13fa6.png)

**Comment by jetgeo** [comment]
I believe it is important for drivers and automated vehicles to know that there is a physical barrier present, and not only a restriction. A restriction can be ignored in a case of emergency, a physical barrier may be more tricky to avoid. The two existing types (“`noEntry`” and “`closedToAllVehiclesInBothDirections`”) feels more like restrictions. 
Therefore, my suggestion is that we should have a specific RoadFeatureType “`physicalBarrier`” in the extended `RoadFeatureTypeCode `list. 

Furthermore, we could add a `RoadFeaturePropertyType `“`barrierType`” and a codelist (`BarrierTypeCode`) for types of physical barriers, whose values can be applied as values for the “`barrierType`” property. For example values “`lockedWithKey`”, “`permanent`”,”`restrictedAccess`” etc.


This would give an instance of a physical barrier looking like this:

```
<RoadFeature gml:id="N1">
…
<type xlink:href="http://spec.tn-its.eu/codelists/RoadFeatureTypeCode# physicalBarrier "/>
               <properties>
                        <GenericRoadFeatureProperty>
                            <type xlink:href="http://spec.tn-its.eu/codelists/RoadFeaturePropertyTypeCode# barrierType "/>
                            <valueReference xlink:href=http://spec.tn-its.eu/codelists/BarrierTypeCode#lockedWithKey />
                        </GenericRoadFeatureProperty>
               </properties>
 </RoadFeature>
```

**Comment by tsiobbel** [comment]
I support the addition of a specific RoadFeatureType "physicalBarrier" but we need to reflect a bit more on 1) alignment terminology with e.g. GDF ("Blocked Passage, Blocked Passage Type (Permanently fixed/Removable), DATEXII (AccessRestriction Type), ...  2) directionality ... 3) types of physical barriers, e.g: 

![2020-305b2218_380](https://user-images.githubusercontent.com/38555630/154970878-e8fc0702-ec49-4e87-b4f1-3eb1d5f362eb.jpg)
Image source: [https://actionhandling.co.uk/our-store/height-restriction-barriers1](url)

**Comment by jetgeo** [comment]
I agree with the need to harmonize with other specifications. And I believe aligning with INSPIRE is most urgent, as it is an overlap on data types provided with INSPIRE and TN-ITS.  

- INSPIRE `AccessRestriction`  (https://inspire.ec.europa.eu/featureconcept/AccessRestriction) covers both legal and physical restrictions, with the value `physicallyImpossible` as the only possible value for physical barriers (https://inspire.ec.europa.eu/codelist/AccessRestrictionValue).
- Personally, I like the term `BlockedPassage` from GDF better than `AccessRestriction`. But on the other hand, `AccessRestriction` is more generic, and one can refer to the INSPIRE concept .

One option can be to duplicate the INSPIRE model into the TN-ITS model, and in addition add `barrierType`  as a FeaturePropertyType for an optional and more detailed classification, plus a code list for barrier types  (`lockedWithKey`, `permanent`, `restrictedAccess` etc.). That way, the model would be harmonized with INSPIRE.  

An example feature will then be like this:

```
<RoadFeature gml:id="N1">
…
<type xlink:href="https://inspire.ec.europa.eu/featureconcept/AccessRestriction"/>
               <properties>
                        <GenericRoadFeatureProperty>
                            <type xlink:href="https://inspire.ec.europa.eu/codelist/AccessRestrictionValue"/>
                            <valueReference xlink:href=https://inspire.ec.europa.eu/codelist/AccessRestrictionValue/physicallyImpossible/>
                        </GenericRoadFeatureProperty>
                        <GenericRoadFeatureProperty>
                            <type xlink:href="[http://spec.tn-its.eu/codelists/RoadFeaturePropertyTypeCode#barrierType ](http://spec.tn-its.eu/codelists/RoadFeaturePropertyTypeCode#barrierType)"/>
                            <valueReference xlink:href=http://spec.tn-its.eu/codelists/BarrierTypeCode#lockedWithKey />
                        </GenericRoadFeatureProperty>
               </properties>
 </RoadFeature>
```

**Comment by Mi-I** [comment]
Hi,
We also agree about harmonizing. 
I am worried about more rich raw data and risk of information loss when converting and going 'generic'.  We must make sure that the outcome still can have attribution and classification which are useful and something wanted by industry & users, so they prefer to work with raw data instead.  
I suppose there is room for misunderstanding whatever term we use (eg Inspire or GDF), so as long as it's well defined where to put what, I am fine with it.

As Inspire does not offer incremental change information, if TN-ITS could fill that gap too, it would be awaresome.

**Comment by jetgeo** [comment]
> As Inspire does not offer incremental change information, if TN-ITS could fill that gap too, it would be awaresome.

Exactly, that's my thinking as well. Stable and complete data sets can be provided through INSPIRE, and updates through TN-ITS.

> We must make sure that the outcome still can have attribution and classification which are useful and something wanted by industry & users, so they prefer to work with raw data instead.
> I suppose there is room for misunderstanding whatever term we use (eg Inspire or GDF), so as long as it's well defined where to put what, I am fine with it.

Agree!  I believe the INSPIRE approach is at a level where it is generic but still useful. And by following an existing specification, we can reduce the possibility of misunderstandings. I believe INSPIRE is more acknowledged and used in Europe now than GDF.
